### PR TITLE
Add user control over scalebar numeric formats 

### PR DESCRIPTION
**Pull request by nyalldawson**
Amongst other follow ups to the recent numeric format API support, this exposes the option for controlling the numeric format used by a layout scalebar.

It gives users control over all the formatting properties for the numbers in scalebars, including whether they want thousand separators, decimal places, scientific notation, etc. Very useful in the case of making maps for audiences outside of the current QGIS locale, or when you'd just prefer to vary the style from the locale defaults (e.g. adding thousands separators when the locale default is to hide them)

### CODE REVIEW

File: src/app/layout/qgslayoutscalebarwidget.cpp [hunk 1]
```diff
@@ -50,7 +51,7 @@ QgsLayoutScaleBarWidget::QgsLayoutScaleBarWidget( QgsLayoutItemScaleBar *scaleBa
   connect( mLineJoinStyleCombo, static_cast<void ( QComboBox::* )( int )>( &QComboBox::currentIndexChanged ), this, &QgsLayoutScaleBarWidget::mLineJoinStyleCombo_currentIndexChanged );
   connect( mLineCapStyleCombo, static_cast<void ( QComboBox::* )( int )>( &QComboBox::currentIndexChanged ), this, &QgsLayoutScaleBarWidget::mLineCapStyleCombo_currentIndexChanged );
   connect( mMinWidthSpinBox, static_cast < void ( QDoubleSpinBox::* )( double ) > ( &QDoubleSpinBox::valueChanged ), this, &QgsLayoutScaleBarWidget::mMinWidthSpinBox_valueChanged );
-  connect( mMaxWidthSpinBox, static_cast < void ( QDoubleSpinBox::* )( double ) > ( &QDoubleSpinBox::valueChanged ), this, &QgsLayoutScaleBarWidget::mMaxWidthSpinBox_valueChanged );
```
Comment (nirvn) [inline src/app/layout/qgslayoutscalebarwidget.cpp]: Is this line removal a mistake?

File: src/core/scalebar/qgsnumericscalebarrenderer.h [hunk 2]
```diff
@@ -48,7 +48,7 @@ class CORE_EXPORT QgsNumericScaleBarRenderer: public QgsScaleBarRenderer
   private:
 
     //! Returns the text for the scale bar or an empty string in case of error
-    QString scaleText( double scale ) const;
+    QString scaleText( const QgsScaleBarSettings &settings, double scale ) const;
```
Comment (nirvn) [inline src/core/scalebar/qgsnumericscalebarrenderer.h]: I'd switch the parameter ordering around, having double scale first then const QgsScaleBarSettings &settings. No strong feelings though, feel free to ignore.

File: src/ui/layout/qgslayoutscalebarwidgetbase.ui [hunk 3]
```diff
@@ -753,6 +774,8 @@
   <tabstop>mFillColor2DDBtn</tabstop>
   <tabstop>mStrokeColorButton</tabstop>
   <tabstop>mLineColorDDBtn</tabstop>
+  <tabstop>mLabelHorizontalPlacementComboBox</tabstop>
```
Comment (nirvn) [inline src/ui/layout/qgslayoutscalebarwidgetbase.ui]: Thanks for cleaning up after me. I always forget those tabstops.

File: src/app/layout/qgslayoutscalebarwidget.cpp (hunk 1 above)
Comment (nyalldawson) [reply src/app/layout/qgslayoutscalebarwidget.cpp]: hm yeah

**Comment by nyalldawson** [comment]
Thanks @nirvn, all fixed

**Pull request merged**